salalib/mgraph.cpp

```diff
@@ -1133,10 +1174,50 @@ bool MetaGraph::makeFewestLineMap( Communicator *communicator, int replace )
    int oldstate= m_state;
    m_state &= ~SHAPEGRAPHS;      // Clear axial map data flag (stops accidental redraw during reload) 
 
-   bool retvar = false;
+   bool retvar = true;
 
    try {
-      retvar = m_shape_graphs.makeFewestLineMap(communicator, (replace != 0));
+       // no all line map
+       if (m_all_line_map == -1) {
+          return false;
+       }
+
+       AllLineMap* alllinemap = dynamic_cast<AllLineMap*>(m_shapeGraphs[size_t(m_all_line_map)].get());

```

Comment:
Hmm, not sure I like this.
Isn't an all line map just a shape graph of which you know it's an all line map?
It seems we can have one all line map at max (as we only have on index to it), so why not have that as a special entity, or have the extra data somewhere else?
At least, the result of a `dynamic_cast` needs to be checked for null and the error needs to be clear...